src/software/convert/main_convertMesh.cpp

```diff
@@ -0,0 +1,155 @@
+/**
+ * @brief Convert Mesh
+ */
+int aliceVision_main(int argc, char** argv)
+{
+    // timer initialization
+
+    system::Timer timer;
+
+    // command-line parameters
+
+    std::string verboseLevel = system::EVerboseLevel_enumToString(system::Logger::getDefaultVerboseLevel());
+    std::string inputMeshPath;
+    std::string outputFilePath;
+
+    po::options_description allParams("AliceVision convertMesh");
+
+    po::options_description requiredParams("Required parameters");
+    requiredParams.add_options()
+      ("inputMesh", po::value<std::string>(&inputMeshPath)->default_value(inputMeshPath),
+        "Mesh file path (*.obj, *.mesh, *.meshb, *.ply, *.off, *.stl).")
+      ("output,o", po::value<std::string>(&outputFilePath)->default_value(outputFilePath),
+        "Output file path for the new mesh file (*.obj, *.mesh, *.meshb, *.ply, *.off, *.stl)");
+
+    po::options_description optionalParams("Optional parameters");
+
+    po::options_description logParams("Log parameters");
+    logParams.add_options()
+      ("verboseLevel,v", po::value<std::string>(&verboseLevel)->default_value(verboseLevel),
+        "verbosity level (fatal, error, warning, info, debug, trace).");
+
+    allParams.add(requiredParams).add(optionalParams).add(logParams);
+
+    po::variables_map vm;
+    try
+    {
+        po::store(po::parse_command_line(argc, argv, allParams), vm);
+
+        if(vm.count("help") || (argc == 1))
+        {
+            ALICEVISION_COUT(allParams);
+            return EXIT_SUCCESS;
+        }
+        po::notify(vm);
+    }
+    catch(boost::program_options::required_option& e)
+    {
+        ALICEVISION_CERR("ERROR: " << e.what());
+        ALICEVISION_COUT("Usage:\n\n" << allParams);
+        return EXIT_FAILURE;
+    }
+    catch(boost::program_options::error& e)
+    {
+        ALICEVISION_CERR("ERROR: " << e.what());
+        ALICEVISION_COUT("Usage:\n\n" << allParams);
+        return EXIT_FAILURE;
+    }
+
+    ALICEVISION_COUT("Program called with the following parameters:");
+    ALICEVISION_COUT(vm);
+
+    // set verbose level
+    system::Logger::get()->setLogLevel(verboseLevel);
+
+    // check first mesh file path
+    if(!inputMeshPath.empty() && !fs::exists(inputMeshPath) && !fs::is_regular_file(inputMeshPath))
+    {
+        ALICEVISION_LOG_ERROR("The input mesh file doesn't exist");
+        return EXIT_FAILURE;
+    }
+
+    // check output file path
+    if(outputFilePath.empty())
+    {
+        ALICEVISION_LOG_ERROR("Invalid output");
+        return EXIT_FAILURE;
+    }
+
+    // ensure output folder exists
+    {
+        const std::string outputFolderPart = fs::path(outputFilePath).parent_path().string();
+
+        if(!outputFolderPart.empty() && !fs::exists(outputFolderPart))
+        {
+            if(!fs::create_directory(outputFolderPart))
+            {
+                ALICEVISION_LOG_ERROR("Cannot create output folder");
+                return EXIT_FAILURE;
+            }
+        }
+    }
```

Comment:
maybe we can refactor in function inside alicevision_system to be used in all the other exe

Reply:
It's a good idea. But maybe it's better to do that in another PR with a larger scope ?

Reply:
we can start with this one creating the function in the library and replacing the code in these 2.

I don't know how many other blocks are there like this one in the other binaries but it's a matter of search and replace